### ISSUE TITLE
[Feature Request] Add option to change Super+Q keyboard shortcut

### ISSUE THREAD

**Issue by laurinneff** [adaxi/audio-output-switcher#11]
On Pop!_OS, Super+Q is the default keyboard shortcut to close windows. This extension overrides that keyboard shortcut with its own, which makes it harder to close windows as I have to move my mouse to the close button. From looking at the code, I've found that the extension should add an option in dconf at `/org/gnome/shell/extensions/audio-output-switcher/`, but I can't find an `audio-output-switcher` key in there. It would be great if you could expose the shortcut in GNOME's settings, since that's where most people would look for it

**Comment by laurinneff** [comment]
As a quick workaround commenting out line 109-128 in extension.js probably works, I'll test that now.

**Comment by laurinneff** [comment]
Yep, commenting out those lines does the trick. Not a good fix though, since the shortcut is completely removed and it gets undone when the extension gets updated

**Comment by adaxi** [comment]
For sure you can comment the lines where the keybinding is done. That will get rid of the keybinding.

I'll accept pull requests that allows to customise the keybiding from the GNOME settings.

**Comment by laurinneff** [comment]
I have no idea of extension development on GNOME, so I probably won't be able to help with customizing from GNOME settings.

**Comment by afa7789** [comment]
where is the code stored ?
i want to coment it too ... newbie sorry

**Comment by raibtoffoletto** [comment]
I just started using gnome again and found this extension very useful, in my opinion it should be the default behaviour in gnome, so thanks for that.
However, add keyboard binds without the possibility to change them is a bad design. 
I also think this should something to be worked, or disable globally. 
(I'm very busy, if I can find time to modify the extension to add the functionality I will, but can't promise anything)

**Comment by adaxi** [comment]
I am quite busy at the moment as well, so I have hardly any time to work on this, but I'll accept the pull request.

Ideally it would be nice to be able to configure the shortcut from Gnome Settings, like all other keyboard shortcuts but I don't know if its feasable.

Alternatively a settings page would need to be created for the extension so that the shortcut key can be configured.

**Comment by emko** [comment]
yea on pop os you can't close a window with super + q anymore

**Comment by xoxoMarzy** [comment]
Just go into the schemas folder and change the shortcut and recompile the file to something you want.
I changed it so super+a and disabled the application view in settings (that was the default thing for super+a).
Works fine

**Comment by christianodejesus** [comment]
Hi all, I'm working on this one, I'm trying to make this keybind configurable, because I have the same problem on PopOS.

**Comment by kparal** [comment]
For the last few weeks I was going crazy about Super+Q (which I configured to replace Alt+F4) not working **sometimes** (there's clearly a race when starting gnome-shell, which shortcut "wins"). Finally I nailed it down to this extension. I believe that providing a hardcoded shortcut (which is moreover using such a well-known combo as Super+Q) without any ability to override it (or even know about it) is a very bad decision and should be patched out until a better solution is available.

Anyway, for those who are not willing to wait for something to happen, I found two similar extensions which don't suffer from this problem:
https://extensions.gnome.org/extension/906/sound-output-device-chooser/
https://extensions.gnome.org/extension/858/volume-mixer/
The latter even provides a customizable keyboard shortcut for output device switching (with an overlay indicator when that happens).
I'd honestly probably still prefer this very extension, due to its UI simplicity, but the two above seem to provide satisfactory alternatives (and can be configured heavily themselves).

**Comment by adaxi** [comment]
I have disable the keyboard shortcut code. I am keeping this issue open so that anyone interested can provide a way to configure the shortcut keys.
